packages/attune/src/bin/attune/http.rs

```diff
@@ -0,0 +1,372 @@
+/// let _result = post::<NoBody, _>(&config, "/api/v0/action", &request).await?;
+/// # Ok(())
+/// # }
+/// ```
+#[derive(Debug, Clone, Copy, Default, serde::Serialize)]
+pub struct NoBody;
+
+/// Convenience method for GET requests.
+///
+/// Uses exponential backoff with jitter to retry the request.
+/// ```no_run
+/// # use crate::http::{get, NoBody};
+/// # use crate::config::Config;
+/// # let config = Config::builder().api_token("test").endpoint("http://localhost:8080").build();
+/// # #[derive(Debug, serde::Deserialize)]
+/// # struct SomeResponse;
+///
+/// // Simple GET requests:
+/// let data = get::<SomeResponse, _>(&config, "/api/v0/data", &NoBody).await?;
+///
+/// // GET requests with JSON body:
+/// let body = serde_json::json!({"key": "value"});
+/// let data = get::<SomeResponse, _>(&config, "/api/v0/data", &body).await?;
+/// ```
+#[tracing::instrument]
+pub async fn get<K: DeserializeOwned, T: Serialize + std::fmt::Debug>(
+    ctx: &Config,
+    path: impl AsRef<str> + std::fmt::Debug,
+    body: &T,
+) -> Result<(Option<K>, StatusCode), ErrorResponse> {
+    run_request(async || {
+        ctx.client
+            .get(ctx.endpoint.join(path.as_ref()).unwrap())
+            .json(body)
+            .send()
+            .await
+    })
+    .await
+}
+
+/// Convenience method for POST requests.
+///
+/// ```no_run
+/// # use crate::http::{post, NoBody};
+/// # use crate::config::Config;
+/// # let config = Config::builder().api_token("test").endpoint("http://localhost:8080").build();
+/// # let body = serde_json::json!({});
+/// # #[derive(Debug, serde::Deserialize)]
+/// # struct SomeResponse;
+///
+/// // POST requests with JSON bodies:
+/// let data = post::<SomeResponse, _>(&config, "/api/v0/data", &body).await?;
+///
+/// // POST requests without expecting a response body:
+/// let data = post::<(), _>(&config, "/api/v0/action", &body).await?;
+/// ```
+#[tracing::instrument]
+pub async fn post<K: DeserializeOwned, T: Serialize + std::fmt::Debug>(
+    ctx: &Config,
+    path: impl AsRef<str> + std::fmt::Debug,
+    data: &T,
+) -> Result<(Option<K>, StatusCode), ErrorResponse> {
+    run_request(async || {
+        ctx.client
+            .post(ctx.endpoint.join(path.as_ref()).unwrap())
+            .json(data)
+            .send()
+            .await
+    })
+    .await
+}
+
+/// Convenience method for POST requests with `multipart/form-data`.
+///
+/// ```no_run
+/// # use crate::http::post_multipart;
+/// # use crate::config::Config;
+/// # let config = Config::builder().api_token("test").endpoint("http://localhost:8080").build();
+/// # #[derive(Debug, serde::Deserialize)]
+/// # struct UploadResponse;
+/// let form_data = || reqwest::multipart::Form::new();
+/// let data = post_multipart::<UploadResponse>(&config, "/api/v0/upload", form_data).await?;
+pub async fn post_multipart<T: DeserializeOwned>(
+    ctx: &Config,
+    path: impl AsRef<str> + std::fmt::Debug,
+    data: impl Fn() -> reqwest::multipart::Form,
```

Comment:
Why a closure here? I can't think of any use cases where we wouldn't immediately want to consume the form.